File: backend/src/main/java/com/denuncieagora/denuncie/domain/entities/Report.java
package com.denuncieagora.denuncie.domain.entities;

import com.denuncieagora.denuncie.domain.enums.HateCrimeTypeEnum;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDate;
import java.util.UUID;

@Entity
@Table(name = "report")
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
public class Report {

    @Id @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(nullable = false)
    @Enumerated(EnumType.STRING)
    private HateCrimeTypeEnum about;

    @Column(nullable = false)
    private LocalDate date;

    @Column(nullable = false)
    private String state;

    @Column(nullable = false)
    private String city;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String description;
}

File: DenuncieAgora/app/src/main/java/com/example/denuncieagora/domain/model/Report.kt
package com.example.denuncieagora.domain.model

import android.os.Parcelable
import com.example.denuncieagora.domain.enums.HateCrimeTypeEnum
import kotlinx.parcelize.Parcelize
import java.util.*

@Parcelize
data class Report(
    val id: UUID,
    val identity: String,
    val about: HateCrimeTypeEnum,
    val date: String,
    val state: String,
    val city: String,
    val description: String
): Parcelable
